tests/unit/test_config.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+
+from scc_hypervisor_collector.api import exceptions, ConfigManager
+
+class TestConfigManager:
+
+    @pytest.mark.config('tests/unit/data/config/default/default.yaml', None)
+    def test_single_file_config_load(self, config_manager):
+        config_data = config_manager._load_config()
+        assert config_manager.config_file._str == 'tests/unit/data/config/default/default.yaml'
+        assert not config_manager.config_dir
+        assert config_data.get('credentials')['scc']['username'] == 'default_scc_username'
+        assert config_data.get('credentials')['scc']['password'] == 'default_scc_password'
+
+    @pytest.mark.config(None, 'tests/unit/data/config/empty')
+    def test_empty_dir_config_load(self, config_manager):
+        with pytest.raises(exceptions.NoConfigFilesFoundError) as excinfo:
+            config_manager._load_config()
+        assert 'No config files found' in str(excinfo.value)
+
+    @pytest.mark.config('tests/unit/data/config/nonexisting.yml', None)
+    def test_nonexisting_file_config_load(self, config_manager):
+        with pytest.raises(exceptions.NoConfigFilesFoundError) as excinfo:
+            config_manager._load_config()
+        assert 'No config files found' in str(excinfo.value)
+
+
+    @pytest.mark.config(None, 'tests/unit/data/config/invalid')
+    def test_invalid_dir_config_load(self, config_manager):
+        config_data = config_manager._load_config()
+        # validation is not performed in the _load_config() call and
+        # hence the yaml file is successfully read
+        assert len(config_data) > 0
+
+    @pytest.mark.skip("Skipping because backends from two different files are not merged")
+    @pytest.mark.config(None, 'tests/unit/data/config/lexorder')
+    def test_lexorder_config_load(self, config_manager):
+        config_data = config_manager._load_config()
+        # merges all contents of all the files found in the directory together.
+        # if the same credentials section is defined in multiple files,
+        # it keeps only the section that was last read based
+        # on the lexorder
+        assert len(config_data) > 0
+        assert config_data.get('credentials')['scc']['username'] == 'second_scc_username'
```

Comment:
Do we need both asserts here? Since they are asserting the same field, the first one passing means the second one also is true?

Reply:
Updated to remove the first assertion